lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4804,6 +4806,35 @@ LogicalResult FIRRTLLowering::visitPrintfLike(T op, StringAttr outputFile) {
   return success();
 }
 
+// Stop lowers into a nested series of behavioral statements plus $fatal
+// or $finish.
+LogicalResult FIRRTLLowering::visitStmt(FFlushOp op) {
+  auto clock = getLoweredNonClockValue(op.getClock());
+  auto cond = getLoweredValue(op.getCond());
+  if (!clock || !cond)
+    return failure();
+
+  circuitState.addMacroDecl(builder.getStringAttr("SYNTHESIS"));
+  addToIfDefBlock("SYNTHESIS", std::function<void()>(), [&]() {
+    addToAlwaysBlock(clock, [&]() {
+      circuitState.usedPrintf = true;
+      circuitState.addFragment(theModule, "PRINTF_FD_FRAGMENT");
```

Comment:
Is this unused here?

Reply:
I changed to flush default FD when file name was not provided, so it's used now.